### PR TITLE
feat(Resource Controller): Support resource group filter for list reclamations

### PR DESCRIPTION
**Pull request by apoorva9s14**
## PR summary
To support resource group filter for list reclamations API

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x]  The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x]  Tests for the changes have been added (for bug fixes / features)
- [x]  Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  
A new filter called resource_group_id is added to the list reclamations API

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x]  No

### CODE REVIEW

**Comment by apoorva9s14** [comment]
Signed-off-by: apoorva9s14 [apoorvalaiti16@gmail.com](mailto:apoorvalaiti16@gmail.com)

File: resourcecontrollerv2/resource_controller_v2_examples_test.go [hunk 1]
```diff
@@ -733,11 +734,23 @@ var _ = Describe(`ResourceControllerV2 Examples Tests`, func() {
 			b, _ := json.MarshalIndent(reclamationsList, "", "  ")
 			fmt.Println(string(b))
 
+			listReclamationsOptionsRgID := resourceControllerService.NewListReclamationsOptions()
```
Comment (padamstx) [inline resourcecontrollerv2/resource_controller_v2_examples_test.go:737]: It looks like we now have two invocations of the ListReclamations() operation within this example code fragment.  Let's limit it to just one invocation.  If you can't incorporate the new query param into the existing example code fragment (i.e. use both account id and resource group id when calling ListReclamations() ), then just pick which parameter to use in the example.

**Review by padamstx — CHANGES_REQUESTED** [review]
Everything looks good except that we really should have just a single invocation in the examples code fragment.

File: resourcecontrollerv2/resource_controller_v2_examples_test.go (hunk 1 above)
Comment (padamstx) [reply resourcecontrollerv2/resource_controller_v2_examples_test.go:737]: Also, there was a gosec error during the travis build. You will see in your PR's diff that by re-generating the code you ended up removing a "gosec" comment pertaining to the offending line of code.  Just restore that comment, then you can run `make all` and then `make scan-gosec` in your sandbox to make sure everything is clean after that.

Comment (apoorva9s14) [reply resourcecontrollerv2/resource_controller_v2_examples_test.go]: Thank you for the review @padamstx. I have made the changes suggested by you

**Review by padamstx — APPROVED** [review]
LGTM

**Comment by padamstx** [comment]
@apoorva9s14 Changes look good.  Prior to merging, could you please post a screenshot of the examples and integration tests running with no failures?
Commands:
```
cd resourcecontrollerv2
go test -tags=integration
go test -tags=examples
```
If there's a lot of output, just capture the summary at the end.

**Comment by padamstx** [comment]
@apoorva9s14 Should I merge this PR now?   If yes, please post screenshots showing a clean run of the int tests and examples as mentioned above.  Thanks!

**Pull request merged**